### ISSUE TITLE
local search step omitted in MCCGA

### ISSUE THREAD

**Issue by jbytecode** [jmejia8/Metaheuristics.jl#28]
Let's suppose we have that scenario

```julia
julia> using Metaheuristics

julia> myf(x) = abs(x[1]-pi) + abs(x[2]-exp(1))
myf (generic function with 1 method)

julia> bounds = [-50.0 -50; 50 50]
2×2 Matrix{Float64}:
 -50.0  -50.0
  50.0   50.0

julia> result = optimize(myf, bounds, MCCGA(use_local_search=true))
+=========== RESULT ==========+
  iteration: 2588
    minimum: 0.017063
  minimizer: [3.124997854232788, 2.71875]
    f calls: 5175
 total time: 0.0736 s
stop reason: Other stopping criteria.
+============================+

```

I have tried that example without the package Optim installed. It is supposed the library to warn "The package Optim must be installed" but it silently report a premature converged result. When we import Optim we get a better solution:

```julia
julia> import Optim

julia> result = optimize(myf, bounds, MCCGA(use_local_search=true))
+=========== RESULT ==========+
  iteration: 2715
    minimum: 1.53782e-08
  minimizer: [3.1415926408130987, 2.7182818258575887]
    f calls: 5540
 total time: 1.8034 s
stop reason: Other stopping criteria.
+============================+
```

The function uses Optim if it is available, however, if it is not the case, it silently reports the solution obtained in previous stages.

**Comment by jmejia8** [comment]
Thanks for reporting this behavior of the package. The next update will display a message on whether the local search has not been performed.

Note that this mainly occurs if the following stopping criterion is satisfied (without Optim):
https://github.com/jmejia8/Metaheuristics.jl/blob/adc58725ce9b19e3f433705501fd50282f4f387d/src/algorithms/MCCGA/MCCGA.jl#L190

**Comment by jmejia8** [comment]
Solved in [db18dd2](https://github.com/jmejia8/Metaheuristics.jl/commit/db18dd26774db96762e8c5e2861d41d358ed5379)

**Comment by jbytecode** [comment]
many thanks
